src/channel.ts

```diff
@@ -113,6 +119,8 @@ export function getSupportedRole(channel: Channel): SupportedRole {
   switch (channel) {
     case X:
     case Y:
+    case X2:
```

Comment:
I'm not 100% sure but I think X2 Y2 should not support `dimension`.  

(Can you think of a case where you want to map group-by variable of aggregate plot to X2/Y2?)